pundit.gemspec

```diff
@@ -19,12 +20,4 @@ Gem::Specification.new do |gem|
   gem.require_paths = ["lib"]
 
   gem.add_dependency "activesupport", ">= 3.0.0"
-  gem.add_development_dependency "activemodel", ">= 3.0.0"
```

Comment:
@olleolleolle This PR looks solid, thanks a lot.

Could you reiterate on these lines, a hint on why we can remove these development dependencies would be nice.

Reply:
@olleolleolle I'm doing some work on the gem and trying to follow this thread. Can you explain how the ruby version example you mention above is any different when specifying gems in the `Gemfile`? My understanding is that a Gemfile can only specify one ruby version. 

Also, while we want the gem to be compatible with as many ruby versions as possible, for the purposes of _development_ of the gem I'm much more open to supporting a smaller set of ruby versions.

Reply:
Adding some notes, in order to clarify:

One of the key things is that a Ruby library repository often git-ignores its `Gemfile.lock`, so that that file is re-created at the `bundle install` invocation.

The single RUBY_VERSION becoming a single Ruby version number, and informs this CI run's creation of its `Gemfile.lock`.

Reply:
@olleolleolle Thanks for responding so quickly 😄  

The Bundler docs seem to recommend committing `Gemfile.lock`: presumably so that all developers of the gem are using the same versions: https://bundler.io/guides/creating_gem.html#testing-our-gem - I'm considering doing that.

Is the point you're making that somehow a lockfile can be different under different ruby versions? I don't know what the mechanism for that would be for that, but I can believe that it's true. What I still don't quite understand is why it would be any different with the gems specified in the Gemfile rather than the gemspec - sorry if I'm missing something obvious.

Reply:
The subtle note is: apps should def. check in their Gemfile.lock. Libraries, it's more subtle, there. Libraries want to support the widest possible matrix of Ruby versions.

- When we ask for `gem "todeloo", "~> 2.2"`, we're placing a constraint on the "solution" of the `Gemfile.lock` (i.e. its resulting content).
- When we ask for a specific `ruby 2.2.2`, we're placing _another_ constraint on the solution. [Docs section about this directive](https://bundler.io/v2.0/man/gemfile.5.html#RUBY)
- By writing `ruby RUBY_VERSION`, we make that constraint "the currently-running Ruby"

Reply:
@olleolleolle I totally agree that libraries in _usage_ want to support many ruby versions, but is the same true for libraries in _development_? Note that all the gems we're talking about here are development dependencies, so they should have no impact at all on _users_ of the gem, right? Only on _developers_ of the gem. 

Again, the docs that I linked to seem to be Bundler themselves specifically saying that `Gemfile.lock` should be checked in _for gems_, since it only impacts gem maintainers, not gem users. Or am I reading it wrong?

(Thanks for taking the time to go down this rabbithole with me by the way - I really appreciate it)

Reply:
I found a better link: https://bundler.io/man/bundle-install.1.html#THE-GEMFILE-LOCK

> When Bundler first shipped, the Gemfile.lock was included in the .gitignore file included with generated gems. Over time, however, it became clear that this practice forces the pain of broken dependencies onto new contributors, while leaving existing contributors potentially unaware of the problem. Since bundle install is usually the first step towards a contribution, the pain of broken dependencies would discourage new contributors from contributing. As a result, we have revised our guidance for gem authors to now recommend checking in the lock for gems.

Reply:
The rabbit-hole expands! Thanks for taking me to task and re-read the section.

_Recommended praxis has shifted & been documented!_

Reply:
@olleolleolle I'm not sure I understand your last comment?

Reply:
Thanks for telling me about the updated documentation! I wasn't aware.